applications/informationmodel-cat/src/main/java/no/fdk/imcat/service/APIHarvest.java

```diff
@@ -0,0 +1,152 @@
+package no.fdk.imcat.service;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.core.util.DefaultPrettyPrinter;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.ObjectWriter;
+import com.fasterxml.jackson.databind.SerializationFeature;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import io.swagger.v3.oas.models.media.Schema;
+import no.dcat.client.registrationapi.ApiRegistrationPublic;
+import no.fdk.imcat.model.InformationModel;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+
+import static no.fdk.imcat.service.InformationmodelHarvester.API_TYPE;
+import static no.fdk.imcat.service.InformationmodelHarvester.RETRY_COUNT_API_RETRIEVAL;
+
+/**
+ * Harvest our own APIS so we can create Information Models from those that have schema
+ */
+@Service
+public class APIHarvest {
+
+    private static final Logger logger = LoggerFactory.getLogger(APIHarvest.class);
+    private ObjectMapper mapper;
+    private RegistrationApiClient registrationApiClient;
+
+    @Autowired
+    APIHarvest(RegistrationApiClient client, ObjectMapper mapper) {
+        this.mapper = mapper;
+        this.registrationApiClient = client;
+    }
+
+    private static String transformRef(String reference) {
+        if (reference == null || reference.trim().isEmpty()) {
+            return reference;
+        }
+
+        return reference.replaceFirst("components/schemas", "definitions");
+    }
+
+    InformationModel getInformationModel(InformationModelHarvestSource source) {
+        InformationModel newModel = new InformationModel();
+        newModel.setHarvestSourceUri(source.URI);
+        ObjectWriter writer = mapper.writer(new DefaultPrettyPrinter());
+        try {
+            mapper.enable(SerializationFeature.INDENT_OUTPUT);
+            newModel.setSchema(writer.writeValueAsString(source.rootToSchema));
```

Comment:
Det er vanskelig å forstå, hvorfor lagrer du schema midlertidlig inni source record. Tror do at source er noe generellt at noen ganger det allerede har json-schema i?

Reply:
Jeg går ut i fra at når vi henter modeller fra andre, så vil vi oppleve at det må to operasjoner til for å hente dem, en GetListOfModel og deretter en GetSingleInformationModel per modell. Jeg går nå i utgangspunktet ut i fra at vår måte å hente apier på (en GetList som gir alle apier) ikke er typisk.